_source/_posts/2022-06-14-simple-crud-react-and-spring-boot.md

```diff
@@ -0,0 +1,1348 @@
+import java.util.Set;
+
+@Data
+@NoArgsConstructor
+@AllArgsConstructor
+@Builder
+@Entity
+public class Event {
+
+    @Id
+    @GeneratedValue
+    private Long id;
+    private Instant date;
+    private String title;
+    private String description;
+    @ManyToMany
+    private Set<User> attendees;
+}
+```
+
+And, a `User.java` class.
+
+```java
+package com.okta.developer.jugtours.model;
+
+import lombok.AllArgsConstructor;
+import lombok.Data;
+import lombok.NoArgsConstructor;
+
+import javax.persistence.Entity;
+import javax.persistence.Id;
+
+@Data
+@NoArgsConstructor
+@AllArgsConstructor
+@Entity
+public class User {
+
+    @Id
+    private String id;
+    private String name;
+    private String email;
+}
+```
+
+Create a `GroupRepository.java` to manage the group entity.
+
+```java
+package com.okta.developer.jugtours.model;
+
+import org.springframework.data.jpa.repository.JpaRepository;
+
+import java.util.List;
+
+public interface GroupRepository extends JpaRepository<Group, Long> {
+    Group findByName(String name);
+}
+```
+
+To load some default data, create an `Initializer.java` class in the `com.okta.developer.jugtours` package.
+
+```java
+package com.okta.developer.jugtours;
+
+import com.okta.developer.jugtours.model.Event;
+import com.okta.developer.jugtours.model.Group;
+import com.okta.developer.jugtours.model.GroupRepository;
+import org.springframework.boot.CommandLineRunner;
+import org.springframework.stereotype.Component;
+
+import java.time.Instant;
+import java.util.Collections;
+import java.util.stream.Stream;
+
+@Component
+class Initializer implements CommandLineRunner {
+
+    private final GroupRepository repository;
+
+    public Initializer(GroupRepository repository) {
+        this.repository = repository;
+    }
+
+    @Override
+    public void run(String... strings) {
+        Stream.of("Seattle JUG", "Denver JUG", "Dublin JUG",
+                "London JUG").forEach(name ->
+                repository.save(new Group(name))
+        );
+
+        Group djug = repository.findByName("Seattle JUG");
+        Event e = Event.builder().title("Micro Frontends for Java Developers")
+                .description("JHipster now has microfrontend support!")
+                .date(Instant.parse("2022-09-13T17:00:00.000Z"))
+                .build();
+        djug.setEvents(Collections.singleton(e));
+        repository.save(djug);
+
+        repository.findAll().forEach(System.out::println);
+    }
+}
+```
+
+**TIP:** If your IDE has issues with `Event.builder()`, it means that you need to turn on annotation processing and/or install the Lombok plugin. I had to uninstall/reinstall the Lombok plugin in IntelliJ IDEA to get things to work.
+
+If you start your app (using `./mvnw spring-boot:run`) ...
+
+... it will fail because Spring Boot 2.7.0 forces H2 v2.0. The H2 2.0 ecosystem doesn't seem like it's quite ready for prime-time, so I recommend you downgrade to H2 version `1.4.200` in your `pom.xml`.
```

Comment:
```suggestion
... it will fail because Spring Boot 2.7.0 forces H2 v2.0. The H2 2.0 ecosystem doesn't seem like it's quite ready for prime time, so I recommend you downgrade to H2 version `1.4.200` in your `pom.xml`.
```
two words no hyphen, per dictionary